ath79: Add support for LONGDATA APS256

mrroyh opened the pull request:
Forum discussion : [https://forum.openwrt.org/t/aps-256va-help-for-identification/143653/52](url)

Specification:
Power: 12-36V input via 5,5/2,1 DC barrel jack, or Micro USB-B
CPU: Atheros AR9344 rev 2
RAM: 128MB
Flash: 16MB
WI-Fi 2.4GHz
Fast Ethernet: 1 WAN and 2 LAN
USB: 2 x USB-A, 1 x micro-USB-B (for power input)
WWAN: 3G modem via extended mini-PCIE form factor (can be replaced with Wifi 5GHz card)

The device come with custom openwrt BB an CC.

Because of limited LAN port, I disable GMAC0, so the WAN port can be connected to GMAC1 and function as LAN port as well. 

Enable ssh access and Backup:
1. open router admin page via LAN cable
2. browse 192.168.111.1:8000
3. login with password 123456
4. click wifi icon on top menu
5. change the path at the end of the url (after random hash) with /admin/system/flashops
    it will looks like this:
    http://192.168.111.1:8000/cgi-bin/luci/;stok=29698152cf64c980177a04f86c99ea0d/admin/system/flashops
    (the hash after "stok=" will be different)
6. restore the config with this modified backup (can be created manually by changing dropbear config to allow ssh)
    https://drive.google.com/file/d/1Vs-k7DHBSRZFfkxv1cMOmgAPZfB-RUen/view?usp=sharing
7. now you can login to ssh with root user and 123456 password, and backup all partition and upgrade firmware

**!!! BACKUP EVERY PARTITION !!!**

Flashing instructions:

~~1. download uboot from hackpascal breed-ar934x-blank.bin[https://breed.hackpascal.net/breed-ar934x-blank.bin](url)~~
~~2. download release firmware as firmware.bin~~
~~3. flash uboot:~~
    ~~`mtd write /tmp/breed-ar934x-blank.bin /dev/mtdX`~~
    ~~replace X with uboot partition (it should be mtd0)~~
~~6. enter breed bootloader you can use breedenter software from pc or erase firmware partition using command~~
    ~~`mtd erase /dev/mtdX`~~
    ~~replace X with firmware partition.~~
~~9. to show partition list use command~~
    ~~`cat /proc/mtd`~~
~~11. flash firmware from breed~~

Flash directly from factory web interface accessed from "Enable ssh access" step 5

musashino205 commented:
Please use the OEM bootloader for contribution to the official OpenWrt repository instead of third party one.

mrroyh commented:
ok, i have change the patch to produce image that can be flash directly from factory web interface, the original firmware is openwrt and no need to flash external bootloader, i dont know why i change the firmware type to tplink since i was learning to port new device.

musashino205 commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,180 @@
+&eth1 {
+	status = "okay";
+
+	nvmem-cells = <&macaddr_art_1002 0>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <(-1)>;
```
`mac-address-increment` is not supported anymore, please use `nvmem-cells` for that instead.

```suggestion
	nvmem-cells = <&macaddr_art_1002 (-1)>;
	nvmem-cell-names = "mac-address";
```

musashino205 commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,180 @@
+&wmac {
+	status = "okay";
+
+	nvmem-cells = <&macaddr_art_1002 0>, <&cal_art_1000>;
+	nvmem-cell-names = "mac-address", "calibration";
```
The mac address is embedded to the caldata and this `"mac-address"` cell may be unnecessary.

mrroyh commented:
edited and compiled locally and mac is ok

hauke commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,179 @@
+		compatible = "gpio-leds";
+
+		wlan2g {
+			label = "blue:wlan2g";
```
Please use functions:
```suggestion
			function = LED_FUNCTION_WLAN;
			color = <LED_COLOR_ID_BLUE>;
```

hauke replied:
same for the others.

hauke commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,179 @@
+};
+
+&builtin_switch {
+	/delete-property/ qca,phy4-mii-enable; /* need to delete this to make all 5 FE ports work */
+/*	pinctrl-names = "default";
+	pinctrl-0 = <&pmx_leds_switch>;*/
```
This looks strange. You probably have to request the correct pins. Your change will probably not reconfigure something and keep the settings from the bootloader.

mrroyh replied:
in the original  firmware wan port is connected to GMAC0, because of limited port number i change it to connect to ethernet switch which is connected to GMAC1.

hauke replied:
When you use the device as a router it is better to connect the WAN port to a different MAC, because the WAN port would then get its own 1GBit/s link and doe snot have to share it with the LAN ports.

mrroyh replied:
in the default firmware the wan port speed only 100mbps, that is why i think its better just to connect it to switch directly. how to connect both gmac to the switch so the user can choose to connect which port to which gmac like this picture(ilustration)
![image](https://github.com/user-attachments/assets/b1d558bd-3181-4d79-8b05-5f09dd41956a)

hauke replied:
ok, if this increases the speed from 100MBit/s to 1GBit/s its is fine.

mrroyh commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,179 @@
+		};
+		
+		lan2 {
+			label = "blue:lan2";
```
@hauke if lan1 function is LED_FUNCTION_LAN, what lan2 function set to?

hauke replied:
See https://github.com/openwrt/openwrt/pull/17939#discussion_r1976398410

musashino205 commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,176 @@
+		};
+
+		lan1 {
+			function = "lan1";
```
You can use the followings:

- `function = LED_FUNCTION_LAN;`
- `function-enumerator = <1>;`

Then, this LED will be appeared as `blue:lan-1`.

hauke commented on target/linux/ath79/dts/ar9344_longdata_aps256.dts:
```diff
@@ -0,0 +1,179 @@
+		};
+
+		led_system: system {
+			label = "blue:system";
```
We use the `LED_FUNCTION_STATUS` function often for system.

mrroyh committed:
ath79: add support for Longdata APS256

Forum discussion : https://forum.openwrt.org/t/aps-256va-help-for-identification/143653/52

Specification:
Power: 12-36V input via 5,5/2,1 DC barrel jack, or 5V Micro USB-B
CPU: Atheros AR9344 rev 2
RAM: 128MB
Flash: 16MB
WI-Fi: 2.4GHz
Fast Ethernet: 1 WAN and 2 LAN
USB: 2 x USB-A, 1 x micro-USB-B (for power input)
WWAN: 3G modem via extended mini-PCIE form factor (can be replaced with Wifi 5GHz card)

The device come with custom openwrt BB an CC.

Because of limited LAN port, I disable GMAC0, so the WAN port can be connected to GMAC1 and function as LAN port as well.

Enable ssh access and Backup:

1.  open router admin page via LAN cable
2.  browse 192.168.111.1:8000
3.  login with password 123456
4.  click wifi icon on top menu
5.  change the path at the end of the url (after random hash) with /admin/system/flashops
        it will looks like this:
        http://192.168.111.1:8000/cgi-bin/luci/;stok=29698152cf64c980177a04f86c99ea0d/admin/system/flashops
        (the hash after "stok=" will be different)
6.  restore the config with this modified backup (can be created manually by changing dropbear config to allow ssh)
        https://drive.google.com/file/d/1Vs-k7DHBSRZFfkxv1cMOmgAPZfB-RUen/view?usp=sharing
7.  now you can login to ssh with root user and 123456 password, and backup all partition and upgrade firmware

!!! BACKUP EVERY PARTITION !!!

Flashing instructions:
- Flash directly from factory web interface accessed from "Enable ssh access" step 5

Signed-off-by: Roy H <roy@altbytes.com>
Link: https://github.com/openwrt/openwrt/pull/17939
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
